#!/usr/bin/env python
PACKAGE = "auv_mapping"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gate_to_slalom_entrance_distance", double_t, 0, "Distance from gate exit to slalom entrance", 1.6, -15, 15)
gen.add("second_slalom_offset", double_t, 0, "Lateral offset for the second slalom waypoint", 0.08, -10, 10)
gen.add("third_slalom_offset", double_t, 0, "Lateral offset for the third slalom waypoint", 0, -10, 10)
gen.add("vertical_distance_between_slalom_clusters", double_t, 0, "Vertical distance between slalom waypoints", 2.0, -5, 5)
gen.add("slalom_entrance_backed_distance", double_t, 0, "Distance for the backed slalom entrance frame", 0.0, -5, 5)
gen.add("faruk_y_offset", double_t, 0, "Faruk's Y offset", 0.0, -15, 15)
gen.add("faruk_x_offset", double_t, 0, "Faruk's X offset", 0.0, -15, 15)
exit(gen.generate(PACKAGE, "auv_mapping", "SlalomTrajectory"))
